nvme_gw_server.py

```diff
@@ -185,24 +217,19 @@ def start_spdk(self):
         })
 
         try:
-            self.spdk_rpc_client = self.spdk_rpc.client.JSONRPCClient(
-                spdk_rpc_socket,
-                None,
-                timeout,
-                log_level=log_level,
-                conn_retries=conn_retries,
-            )
-            self.spdk_rpc_ping_client = self.spdk_rpc.client.JSONRPCClient(
-                spdk_rpc_socket,
-                None,
-                timeout,
-                log_level=log_level,
-                conn_retries=conn_retries,
+            self.spdk_rpc_client_pool = SharedPool(
+                lambda _ : self.spdk_rpc.client.JSONRPCClient(
```

Comment:
We need to think about what should happen if the rpc client fails, e.g. when the connection is dropped. Should we recreate new rpc clients in the pool on exceptions or do we fail the gateway?

Reply:
Do you think there could be a case one one connection (client) is dropped by the spdk, while others are still working?

I expect if one fails the others will fail too. And we have the "spdk health check" (ping) loop, which will terminate the gateway if the ping fails.

Reply:
#24